### PR TITLE
Enable multiple deployers in `skaffold.yaml`

### LINKED ISSUE #2875: Unable to use 2 or more deploy tools in 1 skaffold file
Following [k8s docs](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret-from-generator) want to create secrets with `kustomize` and use `kubectl` for other configs.

### Expected behavior

No error

### Actual behavior

```bash
$ skaffold diagnose
FATA[0000] creating runner: invalid skaffold config: only one element in set deploy can be set. got HelmDeploy and KubectlDeploy
```

### Information

- Skaffold version: v0.38.0
- Operating system: Mac Os 10.14.6
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta14
kind: Config
build:
  tagPolicy:
    envTemplate:
      template: "{{.RELEASE}}-{{.IMAGE_NAME}}"
  artifacts:
    - image: core
      context: core
      docker:
        dockerfile: docker/development/Dockerfile
    - image: pubsub_emulator
      context: ../pubsub_emulator
      docker:
        dockerfile: ../pubsub_emulator/Dockerfile
deploy:
  kubectl:
    manifests:
      - support/rbac/endpoints.yaml
      - support/pubsub_emulator/k8s.yaml
  kustomize:
    path: support/env_vars
  helm:
    releases:
      - name: core
        chartPath: core/charts/poker-core
        valuesFiles:
          - core/charts/core/values.yaml
          - core/charts/core/values.development.yaml
```

### PR DESCRIPTION
**Pull request by corneliusweig**
Relates to #2875, #3292

Fixes #2875, #3292

Should merge before : n/a

Should merge after : #3391, ~~#3390~~

**Description**

Some users find it limiting that they can only configure a single deployer in Skaffold (#2875, #3292). This PR enables using multiple deployers.

**User facing changes**

Users get the possibility to configure multiple deployers.

> :warning: When merging profiles into the base pipeline, Skaffold used to wipe other deployers. This is no longer the case, because Skaffold now handles more than one deployer.

**Before**

`skaffold.yaml` allows to configure _exactly one_ deployer (`kubectl`, `kustomize`, `helm`).

**After**

Users can configure all three deployers at once. Iow, the `oneOf=deployer` constraint in `skaffold.yaml` is dropped.

**Next PRs.**

n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Mentions any output changes.
- [x] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [x] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

```
- The deployer section in `skaffold.yaml` now accepts multiple deployers in a single pipeline.
- When applying profiles, deployers in the base profile no longer get wiped when merging in the deployer from the profile.
```

### CODE REVIEW

**Comment by corneliusweig** [comment]
I went with the approach to simply drop the `oneOf` constraint. Another option would have been to add a list of deployers in the `deploy` section, like so:

```yaml
deploy:
  kubeContext: something
  deployers:
  - kustomize: {}
  - kubectl: {}
  - helm: {}
  - ...
```

However, the nested `deploy.deployers` looks very weird and it is not clear if a list of deployers would even be desirable.

**Comment by corneliusweig** [comment]
I've finally added the missing integration test, so it's no longer WIP :)

File: docs/content/en/docs/pipeline-stages/deployers.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Skaffold supports the following tools for deploying applications:
 The `deploy` section in the Skaffold configuration file, `skaffold.yaml`,
 controls how Skaffold builds artifacts. To use a specific tool for deploying
 artifacts, add the value representing the tool and options for using the tool
-to the `deploy` section.
+to the `deploy` section. You may even use different tools at the same time.
```
Comment (balopat) [inline docs/content/en/docs/pipeline-stages/deployers.md]:
```suggestion
to the `deploy` section. You may even use a combination of each available deployment approach at the same time.
```

**Comment by balopat** [comment]
I just rebased it - hope that's fine!

File: docs/content/en/docs/pipeline-stages/deployers.md (hunk 1 above)
Comment (corneliusweig) [reply docs/content/en/docs/pipeline-stages/deployers.md]: I'll not apply this suggestion right now, to not interfere with the kokoro run. But please go ahead and apply this if you think it's ready for merging!

**Comment by corneliusweig** [comment]
> I just rebased it - hope that's fine!

Yes of course!

File: docs/content/en/docs/pipeline-stages/deployers.md (hunk 1 above)
Comment (balopat) [reply docs/content/en/docs/pipeline-stages/deployers.md]: I just pushed it - will merge as soon as kokoro is done again :)

**Pull request merged**